front/app/containers/ProjectsShowPage/timeline/VotingResults/index.tsx

```diff
@@ -6,22 +6,53 @@ import useIdeas from 'api/ideas/useIdeas';
 // components
 import { Box, useBreakpoint } from '@citizenlab/cl2-component-library';
 import VotingResultCard from './VotingResultCard';
+import { VotingMethod } from 'api/phases/types';
 
 interface Props {
   phaseId: string;
+  votingMethod: VotingMethod;
 }
 
-const VotingResults = ({ phaseId }: Props) => {
+const VotingResults = ({ phaseId, votingMethod }: Props) => {
   const { data: ideas } = useIdeas({
     phase: phaseId,
-    sort: 'votes_count',
+    sort: votingMethod === 'budgeting' ? 'baskets_count' : 'votes_count',
   });
-
   const smallerThanPhone = useBreakpoint('phone');
 
+  if (!ideas) return null;
+
   const getMx = (i: number) =>
     smallerThanPhone ? {} : i % 2 ? { ml: '8px' } : { mr: '8px' };
 
+  const getRanking = (
+    index: number,
+    attributeName: 'baskets_count' | 'votes_count'
+  ): number => {
+    let currentRank = 1;
+    let prevCount = ideas.data[0].attributes[attributeName];
+
+    for (let i = 0; i <= index; i++) {
+      const currentCount = ideas.data[i].attributes[attributeName];
+
+      if (
+        typeof prevCount === 'number' &&
+        typeof currentCount === 'number' &&
```

Comment:
Because we don't differentiate the type of ideas based on the voting method, we unfortunately have to do these checks, even though the counts should already be a number. TS can't infer this based on our type definitions.

Reply:
Hmm I'm not sure about this algorithm. I tried to recreate it in the browser somewhat:
![Screenshot 2024-02-06 at 10 18 55](https://github.com/CitizenLabDotCo/citizenlab/assets/13822005/35219cf2-dafc-4390-a1d2-831c40e9f046)

I think usually, when you have ties, you continue counting after the tie as if there was no tie. So in this example I would expect `getRanking(3)` to be `4` instead of `3`. Does that make sense?

I would recommend the following algorithm instead because it's more type safe, performant, correct, easier to test and easier to understand (IMO):

```ts
// outside of component
const getRanks = (counts: number[]) => {
    let currentRank = 1;
    const ranks = [];
    
    for (let i = 0; i < counts.length; i++) {
        if (i === 0) {
            ranks.push(currentRank);
        } else {
            const count = counts[i];
            const prevCount = counts[i - 1];

            if (count < prevCount) {
                currentRank = i + 1;
                ranks.push(currentRank);
            } else {
                ranks.push(currentRank);
            }
        }
    }

    return ranks;
}

const getCounts = ((
  ideas: IIdeas, 
  attributeName: 'baskets_count' | 'votes_count'
) => ideas.data.map((idea) => idea.attributes[attributeName] ?? 0);

// inside of component
const budgeting = votingMethod === 'budgeting';
const counts = getCounts(ideas, budgeting ? 'baskets_count' : 'votes_count');
const ranks = getRanks(counts);

[...]

<VotingResultCard rank={ranks[i]} />
```

Here's how it behaves: 
![Screenshot 2024-02-06 at 10 26 21](https://github.com/CitizenLabDotCo/citizenlab/assets/13822005/1ac1ba9f-1a39-4d8c-a418-d23aad2e5abc)